app/models/translation.rb

```diff
@@ -4,4 +4,48 @@ class Translation < ActiveRecord::Base
 	attr_accessible :user, :video
 
 	validates_presence_of :video
+
+	def video
+		Video.find(self.video_id)
+	end
+
+	def self.create_and_save_new_translation(video, user, srt)
+		self.verify_file(srt)
+		root_folder_name = "public/srts/"
+		self.make_folder(root_folder_name)
+		video_folder_name = "#{root_folder_name}#{video.id}"
+		self.make_folder(video_folder_name)
+		self.remove_srt_if_exists(video, user, video_folder_name)
+		translation = Translation.create(:video => video, :user => user)
+		file_name =  "#{translation.id}.srt"
+    	path = File.join(video_folder_name, file_name)
+    	File.open(path, "wb") { |f| f.write(srt.read) }
+		return translation
+	end
+
+	def self.remove_srt_if_exists(video, user, video_folder_name)
+		other_records = Translation.where(:user_id => user.id, :video_id => video.id)
+		other_records.each do |record|
+			other_record_path = File.join(video_folder_name,record.id.to_s)+".srt"
+			if File.exist?(other_record_path)
+				File.delete(other_record_path)
+			end
+			record.destroy
+		end
+	end
+
+	def self.make_folder(folder_name)
```

Comment:
Consider making some of these methods private. These should never be called externally; they're purely helper methods.